### PR TITLE
feat: Handle Password External Inputs RELEASE

### PR DESCRIPTION
**Pull request by tomerlichtash**
## Related Issues

Fixes https://github.com/descope/etc/issues/6314

### CODE REVIEW

**Review by dorsha — DISMISSED** [review]

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 1]
```diff
@@ -658,6 +658,31 @@ class DescopeWc extends BaseDescopeWc {
...
+    }
+
+    const isExternalInput = ele.getAttribute('external-input') === 'true';
+
+    if (!isExternalInput) {
+      return;
+    }
+
+    const origInput = ele.shadowRoot.querySelector('input');
+    const id = `input-${origInput.getAttribute('id')}`;
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:673]: I think we can get the id directly from the element

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 2]
```diff
@@ -799,9 +824,16 @@ class DescopeWc extends BaseDescopeWc {
...
-      setTimeout(() =>
-        updateScreenFromScreenState(this.rootElement, screenState),
-      );
+      setTimeout(() => {
+        updateScreenFromScreenState(this.rootElement, screenState);
+
+        const eles = this.rootElement.querySelectorAll('descope-password');
+
+        // remove existing external inputs
+        document.querySelectorAll('[data-hidden-input="true"]').forEach(ele => ele.remove());
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:833]: Why do we need to do it?

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 3]
```diff
@@ -896,7 +931,8 @@ class DescopeWc extends BaseDescopeWc {
     const id = getFirstNonEmptyValue(formData, idFields);
     const password = getFirstNonEmptyValue(formData, passwordFields);
 
-    if (id && password) {
+    // PasswordCredential not supported in Firefox
+    if (id && password && globalThis.PasswordCredential) {
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:935]: We have the same check 2 lines below, can we remove one?

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 4]
```diff
@@ -658,6 +658,31 @@ class DescopeWc extends BaseDescopeWc {
...
+  #handleDescopePassword(ele: Element) {
+    if (!ele) {
+      return;
+    }
+
+    const isExternalInput = ele.getAttribute('external-input') === 'true';
+
+    if (!isExternalInput) {
+      return;
+    }
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:670]:
```suggestion
    if(ele?.getAttribute('external-input') !== 'true') {
       return;
    }
```

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 5]
```diff
@@ -658,6 +658,31 @@ class DescopeWc extends BaseDescopeWc {
     }
   }
 
+  #handleDescopePassword(ele: Element) {
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:661]: I think it worth a comment about the reason we are doing this

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 3 above)
Comment (tomerlichtash) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts]: Removed

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 2 above)
Comment (tomerlichtash) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts]: To clean up hidden elements after each screen

**Review by nirgur — APPROVED** [review]

**Pull request merged**